python/lsst/ts/mtaos/mtaos.py

```diff
@@ -1857,8 +1873,17 @@ async def pubEvent_m2HexapodCorrection(self) -> None:
 
         model = self.model
         x, y, z, u, v, w = model.m2_hexapod_correction()
+        visitId, extraId = model.get_visit_ids()
         await self.evt_m2HexapodCorrection.set_write(
-            x=x, y=y, z=z, u=u, v=v, w=w, force_output=True
+            x=x,
+            y=y,
+            z=z,
+            u=u,
+            v=v,
+            w=w,
+            visitId=visitId,
+            extraId=extraId,
```

Comment:
This will be a problem when `extraId` is `None`. Integer values cannot be `None`.